### ISSUE TITLE
more than 2 rings for RRP doesn't work

### ISSUE THREAD

**Issue by OndrejHome** [OndrejHome/ansible.pcs-modules-2#22]
Identified in PR #21 that while the `man corosync.conf` mentions that `ringnumber` have to be just unique and doesn't mention any limits.
~~~
ringnumber
       This specifies the ring number for the interface.  When using the redundant ring protocol, each
       interface should specify separate ring numbers to uniquely identify to the membership  protocol
       which interface to use for which redundant ring. The ringnumber must start at 0.
~~~
When configured in CentOS 7.6 with ring2_addr the following error is reported which suggests that either there is limit in CentOS/RHEL version of pacemaker (`pacemaker-1.1.19-8.el7_6.1.x86_64`) or documentation is inaccurate.
~~~
/etc/corosync/corosync.conf
nodelist {
    node {
        ring0_addr: node1-eth0.example.com
        ring1_addr: node1-eth1.example.com
        ring2_addr: node1-eth2.example.com
        nodeid: 1
    }

    node {
        ring0_addr: node2-eth0.example.com
        ring1_addr: node2-eth1.example.com
        ring2_addr: node2-eth2.example.com
        nodeid: 2
    }
}

[MAIN  ] parse error in config: interface ring number 2 is bigger than allowed maximum 1
[MAIN  ] Corosync Cluster Engine exiting with status 8 at main.c:1373.
~~~

This bug is here for tracking on what from above is the case and if there is limitation in CentOS/RHEL then lets make this clear somewhere.

**Comment by OndrejHome** [comment]
OK so the message is printed when we exceed the constant `INTERFACE_MAX` that was `2` (meaning 2 interfaces, with highest interface ID being `1`). This was extended to `8` with introduction of knet and deprecation of RRP in commit https://github.com/corosync/corosync/commit/268cde6ee48ae18004e9de5469f0be97a46e10a0
On Fedora 29 the new corosync 3.x using the knet I can go beyond 2 links like this.
`pcs cluster setup newcluster fastvm-fedora29-38 addr=192.168.22.38 addr=192.168.22.48 addr=192.168.22.58 fastvm-fedora29-39 addr=192.168.22.39 addr=192.168.22.49 addr=192.168.22.59`

~~~
nodelist {
    node {
        ring0_addr: 192.168.22.38
        ring1_addr: 192.168.22.48
        ring2_addr: 192.168.22.58
        name: fastvm-fedora29-38
        nodeid: 1
    }

    node {
        ring0_addr: 192.168.22.39
        ring1_addr: 192.168.22.49
        ring2_addr: 192.168.22.59
        name: fastvm-fedora29-39
        nodeid: 2
    }
}
~~~

In conclusion, the RRP will work only with 2 links, the knet will work with up to 8 and will need changes in code to be usable with `pcs_cluster` module.

**Comment by OndrejHome** [comment]
Changes for proper RRP (upto 8 links with `pcs-0.10`) were incorporated in commit https://github.com/OndrejHome/ansible.pcs-modules-2/commit/887386b48ea77959801d1b451f90786d6c5bbca7 . Closing this issue.
